notes/trees.md

```diff
@@ -0,0 +1,311 @@
+# Trees.scala
+
+## Top level (`meta.Source`, `meta.Pkg`, `meta.Pkg.Object`)
+
+```scala
+"package a".parse[Source].get.structure
+
+Source(stats = List(Pkg(ref = Term.Name("a"), stats = Nil)))
+```
+
+## Declarations (`meta.Decl`)
+
+```scala
+"val a: Int".parse[Stat].get.structure
+
+Decl.Val(
+  mods = Nil,
+  pats = List(Pat.Var(name = Term.Name("a"))),
+  decltpe = Type.Name("Int")
+)
+```
+
+Decl            | Examples
+----------------|-----------
+`Decl.Val`      | `val a: Int`
+`Decl.Var`      | `var a: Int`
+`Decl.Def`      | `def f: Int`
+`Decl.Type`     | `type T`
+
+
+## Definitions (`meta.Defn`)
+
+```scala
+"val a = 1".parse[Stat].get.structure
+
+Defn.Val(
+  mods = Nil,
+  pats = List(Pat.Var(name = Term.Name("a"))),
+  decltpe = None,
+  rhs = Lit.Int(value = 1)
+)
+```
+
+Defn         | Examples
+-------------|-----------
+`Defn.Val`   | `val a = 1`
+`Defn.Var`   | `var a = 1`
+`Defn.Def`   | `def f = 1`
+`Defn.Macro` | `def f = macro impl`
+`Defn.Type`  | `type T = Int`
+`Defn.Class` | `class A`
```

Comment:
`class T`, I think it's important to use different characters for `Term.Name` and `Type.Name`.